liquidator/OneInchExchange.js

```diff
@@ -24,20 +22,53 @@ class OneInchExchange {
     this.oneSplitContract = new web3.eth.Contract(OneSplit.abi, this.oneSplitAddress);
 
     this.toBN = web3.utils.toBN;
+
+    // 1 Split config
+    this.oneInchFlags = 0; // Enables all exchanges
+
+    // Number of pieces source volume could be splitted
+    // (Works like granularity, higly affects gas usage.
+    // Should be called offchain, but could be called onchain
+    // if user swaps not his own funds, but this is still considered
+    // as not safe)
+    // More info: https://github.com/CryptoManiacsZone/1inchProtocol#getexpectedreturn
+    this.oneInchParts = 2;
+  }
+
+  /**
+   * @notice Gets expected returns on one inch
+   * @param {Object} swapArgs - Swap arguments
+   * @param {string} swapArgs.fromToken Address of token to swap from
+   * @param {string} swapArgs.toToken Address of token to swap to.
+   * @param {string} swapArgs.amountWei String amount to swap, in Wei.
+   * @param {Object} options Web3 options to supply to send, e.g.
+   *      { from: '0x0...',
+            value: '1000',
+            gasPrice: '... }
+   */
+  async getExpectedReturn({ fromToken, toToken, amountWei }) {
+    const expectedReturn = await this.oneSplitContract.methods
+      .getExpectedReturn(fromToken, toToken, amountWei, this.oneInchParts, this.oneInchFlags)
+      .call();
+
+    const { returnAmount } = expectedReturn;
+
+    return returnAmount;
   }
 
   /**
    * @notice Swaps token on one inch
    * @param {Object} swapArgs - Swap arguments
    * @param {string} swapArgs.fromToken Address of token to swap from
    * @param {string} swapArgs.toToken Address of token to swap to.
+   * @param {string} swapArgs.minReturnAmountWei Min expected return amount, in Wei.
    * @param {string} swapArgs.amountWei String amount to swap, in Wei.
    * @param {Object} options Web3 options to supply to send, e.g.
    *      { from: '0x0...',
             value: '1000',
             gasPrice: '... }
    */
-  async swap({ fromToken, toToken, amountWei }, options = {}) {
+  async swap({ fromToken, toToken, minReturnAmountWei, amountWei }, options = {}) {
     // Update gasEstimator state
     await this.gasEstimator.update();
```

Comment:
nit also un-related: the clients, like this `oneInchExchange.js` should NOT update the gas estimator. this is a redundant call and it should be expected that the gas estimator is up to date before getting to this point in the code. The main update happens here